lib/src/main/java/com/diffplug/spotless/generic/GenericConstants.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.generic;
+
+import com.diffplug.spotless.SerializableFileFilter;
+
+public final class GenericConstants {
+
+	/**
+	 * {@link LicenseHeaderStep} completely blows apart package-info.java, package-info.groovy and module-info.java.
+	 * This file filter ensures that such files are skipped.
+	 * <p>
+	 * See:
+	 * <ul>
+	 * <li><a href="https://github.com/diffplug/spotless/issues/1">https://github.com/diffplug/spotless/issues/1</a></li>
+	 * <li><a href="https://github.com/diffplug/spotless/issues/270">https://github.com/diffplug/spotless/issues/270</a></li>
+	 * </ul>
+	 */
+	public static final SerializableFileFilter LISENCE_HEADER_UNSUPPORTED_FILES_FILTER = SerializableFileFilter.skipFilesNamed(
+			"package-info.java", "package-info.groovy", "module-info.java");
+
+	private GenericConstants() {}
```

Comment:
- `LISENCE` ought to be spelled `LICENSE`
-  This ought to live in  `LicenseHeaderStep` as `JAVA_FILES_SKIP` or something like that
- It should be exposed as a `public static SerializableFileFilter method();` rather than `public static final SerializableFileFilter field;`.  Changing the value of a public compile-time constant [breaks binary compatibility](https://wiki.eclipse.org/Evolving_Java-based_APIs_2#Evolving_API_interfaces_-_API_fields).

Reply:
Should be fixed now. The field is called `UNSUPPORTED_FILES_FILTER` and method that exposes it is `LicenseHeaderStep#unsupportedFilesFilter()`.  Wdyt?

Would such added `public static final` field become part of the public API? Who is the consumer of this API?

Reply:
> Would such added public static final field become part of the public API?

Yup, if anyone else compiles code against this, the value of the constant gets compiled into their code.  If we change the value of the constant, their code won't change.  Putting it behind a method call fixes this.

> Who is the consumer of this API?

Nobody that I know of.  We can be pretty cavalier about breaking ABI and we'd get away with it, but when it's easy to keep ABI then we might as well.

> `unsupportedFilesFilter()`

The trouble is that `LicenseHeaderStep` is used for lots of different languages, and this constant is Java-specific.  If we need a different filter for Kotlin or Scala or something, how would we name it?  How about `unsupportedJavaFilesFilter()`?

Reply:
I get the ABI point now. Thanks for the explanation.

Filter also excludes `package-info.groovy` that is why I tried not to mention Java in the method name :) Maybe it could be a single filter of all unsupported files?

Reply:
How about `unsupportedJvmFilesFilter()`?  We've open PR's for `c/c++` and `typescript` formatters.  It's easier to build it modular now, and have the option to return the same filter for each method, rather than to build it coupled, and have to deprecate later if we need to uncouple.

Reply:
`unsupportedJvmFilesFilter()` looks good to me. Updated the commit.